dashboard/test/models/pd/application/teacher_application_test.rb

```diff
@@ -390,6 +390,15 @@ class TeacherApplicationTest < ActiveSupport::TestCase
       )
     end
 
+    test 'incomplete application is valid but does not queue an email nor score it' do
+      application = create :pd_teacher_application, :incomplete
+      assert application.valid?
+
+      application.expects(:queue_email).never
+      application.expects(:auto_score!).never
+      application.on_successful_create
+    end
```

Comment:
the above discussion is making me think that this would be better as a controller test -- otherwise, it's not clear (to me) that the create method on the controller isn't doing something else that might send the emails, and it seems important to be sure that emailing and auto-scoring do not happen in this case.

like I said, it's all very subjective... perhaps I am more biased to the clarity of the controller tests because I am still pretty new to this part of the code :-)

Reply:
Yep, that makes sense. Will move

Reply:
Update: will move in a future PR, added a TODO